### ISSUE TITLE
GNU Guix

### ISSUE THREAD

**Issue by phodina** [Shatur/optimus-manager-qt#116]
Hi,

I'm porting this project to GNU Guix.

How should I handle the circle-flags and masalla packages?

I get the following error:
```
starting phase `configure'
source directory: "/tmp/guix-build-optimus-manager-qt-1.6.5.drv-0/source" (relative from build: "../source")
build directory: "/tmp/guix-build-optimus-manager-qt-1.6.5.drv-0/build"
running 'cmake' with arguments ("../source" "-DCMAKE_BUILD_TYPE=RelWithDebInfo" "-DCMAKE_INSTALL_PREFIX=/gnu/store/wig9rzmr7lnzlb52ghjlmi7am8n935kq-optimus-manager-qt-1.6.5" "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE" "-DCMAKE_INSTALL_RPATH=/gnu/store/wig9rzmr7lnzlb52ghjlmi7am8n935kq-optimus-manager-qt-1.6.5/lib" "-DCMAKE_VERBOSE_MAKEFILE=ON")
-- The CXX compiler identification is GNU 10.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /gnu/store/069aq2v993kpc41yabp5b6vm4wb9jkhg-gcc-10.3.0/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:52 (configure_file):
  Permission denied


CMake Error at CMakeLists.txt:53 (configure_file):
  Permission denied


-- Configuring incomplete, errors occurred!
See also "/tmp/guix-build-optimus-manager-qt-1.6.5.drv-0/build/CMakeFiles/CMakeOutput.log".
error: in phase 'configure': uncaught exception:
%exception #<&invoke-error program: "cmake" arguments: ("../source" "-DCMAKE_BUILD_TYPE=RelWithDebInfo" "-DCMAKE_INSTALL_PREFIX=/gnu/store/wig9rzmr7lnzlb52ghjlmi7am8n935kq-optimus-manager-qt-1.6.5" "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE" "-DCMAKE_INSTALL_RPATH=/gnu/store/wig9rzmr7lnzlb52ghjlmi7am8n935kq-optimus-manager-qt-1.6.5/lib" "-DCMAKE_VERBOSE_MAKEFILE=ON") exit-status: 1 term-signal: #f stop-signal: #f>
phase `configure' failed after 0.2 seconds
command "cmake" "../source" "-DCMAKE_BUILD_TYPE=RelWithDebInfo" "-DCMAKE_INSTALL_PREFIX=/gnu/store/wig9rzmr7lnzlb52ghjlmi7am8n935kq-optimus-manager-qt-1.6.5" "-DCMAKE_INSTALL_LIBDIR=lib" "-DCMAKE_INSTALL_RPATH_USE_LINK_PATH=TRUE" "-DCMAKE_INSTALL_RPATH=/gnu/store/wig9rzmr7lnzlb52ghjlmi7am8n935kq-optimus-manager-qt-1.6.5/lib" "-DCMAKE_VERBOSE_MAKEFILE=ON" failed with status 1
builder for `/gnu/store/s4g2m6dr9p7m6dwnx44m8axb6vdyg4fd-optimus-manager-qt-1.6.5.drv' failed with exit code 1
```

Should I point the CMake through environment variables to those packages? I can package them but I'm unsure how to build this package.

Could you please advise?

**Comment by Shatur** [comment]
Hi! It looks like your error is related to these lines:

https://github.com/Shatur/optimus-manager-qt/blob/c9fcb21e715656ad57e218ef13ed832c6e123615/CMakeLists.txt#L52-L53

Which says that you don't have enough rights to write files into this directories.

**Comment by phodina** [comment]
Ok, thanks @Shatur. I'll try to fix this issue

**Comment by Shatur** [comment]
Fixed in 1.6.6, you can download it using submodules.
